File: Erlang/uniao.erl
% c(teste).
% teste:func().

%  uniao: recebe duas listas que não contenham elementos repetidos 
% e retorna uma nova com todos os elementos das duas listas originais 
% (sem repetição)

-module(uniao).

-export([main/2]).

main(Lista1, Lista2) ->
  Aux = Lista2 -- Lista1,
  Result = Lista1 ++ Aux,
  Result.

% uniao:main([3,6,5,7],[2,9,7,5,1]).

File: Haskell/uniao.hs
pertence :: Eq t => t -> [t] -> Bool
pertence a [] = False
pertence a (x:z) = if (a == x) then True
                               else pertence a z

uniao:: Eq t => [t] -> [t] -> [t]
uniao [] l = l
uniao (x:xz) l = if pertence x l then uniao xz l
                                  else x: uniao xz l

main :: IO ()
main = print $ uniao [3,6,5,7] [2,9,7,5,1]

-- ghc -o uniao uniao.hs
-- ./uniao